### ISSUE TITLE
ADKeychainUtil.h file not found (master branch - 2.2.4)

### ISSUE THREAD

**Issue by 5fcgdaeb** [AzureAD/azure-activedirectory-library-for-objc#742]
I have recently upgraded from 2.2.1 to the latest master branch (currently tagged at 2.2.4) and I get compilation error:

#import "ADKeychainUtil.h" (inside ADKeychainTokenCache.m) -> ADKeychainUtil.h file not found

That file is supposed to be under ADAL/src/utils but I have checked to see that it is not there. The file exists but it hasn't been added to the project (.pbxproj file). I have added it myself and then it started working.

Has anybody else come across this? Looks like a very major issue but I wasn't able to see any issues so far reported.

Thanks,
Guven.

**Comment by joeydong** [comment]
Hi Guven,

Yeah I've run into this sometimes when attempting to compile the iOS and Mac targets in the same build run. Does compiling just one or the other work for you?

I think it might just be an Xcode limitation and we should either split the "Build All" target by platform or get rid of it.

**Comment by joeydong** [comment]
Oh I totally didn't read what you wrote. Yeah you might have found the actual source of the problem! Maybe open a PR?

**Comment by RPangrle** [comment]
ADKeychainUtil.h is in the pbxproj: https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/dev/ADAL/ADAL.xcodeproj/project.pbxproj#L517

**Comment by 5fcgdaeb** [comment]
Yes, I can now see the ADKeychainUtil.h and ADKeychainUtil.m files. I am guessing that was due to the XCode indexing issues.

I also had to include ADAL-core library & target instead of the ADALiOS ones to get it working again. The ADAL version upgrades usually end up being breaking updates for us. We are not using CocoaPods for integration, I am sure that is not helping us either.

Closing the issue now.

Thanks,
Guven.
